pkg/listen/listen.go

```diff
@@ -27,34 +30,32 @@ type Listener struct {
 	// Out is an io.Writer to output to.
 	// doctl hint: this should usually be commands.CmdConfig.Out
 	Out io.Writer
-
-	done chan bool
-	stop chan bool
+	// InputCh is a channel to send input to the websocket
+	InCh <-chan []byte
 }
 
 // ListenerService listens to a websocket connection and outputs to the provided io.Writer
 type ListenerService interface {
-	Start() error
-	Stop()
+	Listen(ctx context.Context) error
+	ReadRawStdin(ctx context.Context, stdinCh chan<- string) error
+	MonitorResizeEvents(ctx context.Context, resizeEvents chan<- TerminalSize) error
```

Comment:
This is probably not the right place for these 2 functions. I'm open to ideas on refactoring this.

Reply:
They do feel a bit out of place. A few options:

1. Create a new package named something like `terminal` to hold utils related to `golang.org/x/term` usage
2. The listen package is only used by App Platform related commands. We can rename/redefine it to be more tightly coupled with the App Platform use case rather than trying to be a generic way to listen on a websocket. 
3. Leave it as is. They are used along side the listen package. So living along side that code keeps things quickly understandable

I'm leaning towards the first option. Thoughts?

Reply:
Thanks. I liked option 1 as well. I have updated the PR. I had to add it to `doit` as well to mock it.

I'd have liked to add an integration test as well. But since the tests run in a non-interactive environment, executing this command from the integration test returns this error while it tries to set the terminal to raw mode:
```
Error: error setting terminal to raw mode: operation not supported by device
```